### DISCUSSION TITLE
Shorter binding for props with the same data name

### DISCUSSION THREAD

**Discussion by rinodrummer** [vuejs/rfcs#405] [General]
Hi guys!

I always thought that may be repetitive to re-write the prop value if it has the same name of the data we want to bind, so that's my suggestion: let us use another form of the `v-bind` shorthand for this purpose, for example:

Let's suppose we have a prop called `title` and a data called the same way, what we usually do is this:

```html
<blog-post
    :title="title"
/>
```

As you can see, here we have a redundant information.

My purpose is to let us do this to achieve the same result:

```html
<blog-post
    ::title
/>
```

**Comment by ChristoPy** [comment]
I really like the ideia.  
Sometimes, I wish Vue could reduce this redundancy.  

Svelte and React does this with brackets around the prop name:
```html
<blog-post
    {title}
/>
```

The proposed shorthand is very useful.  
But I think that with double ```::``` could make refactoring a little more difficult.  
Or add too much complexity for new learners (since ```v-bind``` could have more than one shorthand).

So, to reduce the learning curve or refactoring, the new shorthand could be as close as the current ```v-bind``` shorthand.  
```html
<blog-post
    :title
/>
```

**Reply by leopiccionia** [reply]
I like this proposal because it has some symmetry with [HTML boolean attributes](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes), where `checked disabled` is equivalent to `checked="checked" disabled="disabled"`.

**Comment by julie777** [comment]
I'll add my vote for ```v-bind:attr``` or ```:attr``` is equivalent to ```v-bind:attr="attr"``` and ```:attr="attr"``` because when I first saw
```
:title
```
above it seemed obvious that I was binding the title variable to the title attribute. (I am relatively new to Vue, so my reaction is from someone who is not set in her ways and is already tired of typing ```:title="title"``` all the time. I like "script setup" for the same reason.)

**Reply by rinodrummer** [reply]
My idea is suited for every prop of a component, let's suppose you have an `author` string prop, you should automatically bind it with my proposed shorthand like this:

```html
<blog-post
    ::author <!-- Instead of :author="author" -->
/>
```

**Reply by Demivan** [reply]
@rinodrummer It is the same idea. Just instead of `::prop` syntax it is `:prop`. I like it more - there is no need to invent new syntax for this.

**Reply by rinodrummer** [reply]
I know, I used my syntax to make more clear the example I provided in comparison to my original post.

**Comment by edison1105** [comment]
> 
```javascript
<blog-post
    ::author <!-- Instead of :author="author" -->
/>

```
I love this.

**Reply by AndrewBogdanovTSS** [reply]
Why not just `<blog-post :author/>` ?

**Comment by leopiccionia** [comment]
I've just remembered single-colon shorthand was already proposed many times for Vue 2, and Evan wasn't a fan back then.

See examples:
https://github.com/vuejs/vue/pull/2877
https://github.com/vuejs/vue/issues/5835

Quoting one of them:

> Took me some time to respond this because I am not really a fan of this proposal but it wasn't immediately clear to me what turned me off.
>
> Now I realize it's because of the semantic implication of HTML attributes that have no values - they are usually boolean attributes. An attribute with no value indicates the "presence of an attribute" or "truthiness". Implicitly expanding to a binding overloads this perception.
>
> In addition - if a prop in the child component is declared with type Boolean, the presence of the attribute indicates a true value. This would cause `prop` and `:prop` to have very different meanings, and is technically a breaking change

**Comment by darrylnoakes** [comment]
Has this stalled?

Another syntax idea:

```html
<blog-post
    :{title}
/>
```

**Comment by tikudev** [comment]
Since there was no RFC-Discussion open; [I created an RFC regarding this issue](https://github.com/vuejs/rfcs/discussions/508). I proposed a solution which effectively supports this syntax:
```html
<blog-post
    {title}
/>
```

**Reply by matthew-dean** [reply]
I prefer this

**Comment by zhiyuanzmj** [comment]
I created a PR for `vue-macros` https://github.com/vue-macros/vue-macros/pull/493 :

```vue
<script setup lang="ts">
import { ref } from 'vue'
import Child from './child.vue'

const value = ref('')
</script>

<template>
  <input :value />   <!-- <Child :value="value" /> -->

  <Child ::value />  <!-- <Child v-model:value="value" /> -->
</template>
```

**Reply by Kiwasu** [reply]
This is awesome!

**Comment by zhangenming** [comment]
```vue
// how about 
<comp @event></comp>

// equivalent to
<comp @event="event"></comp>
```

**Comment by AndrewBogdanovTSS** [comment]
After I updated to this syntax I start to get eslint errors about unused variables
```
<div v-for="({ image, price, title, url, recToken, rating }, i) in recommendedProducts" :key="i">
  <product-card-recommended :currency="currency[locale]" :title :url :image :price :rating />
</div>
```
```
  17:21  error  'image' is defined but never used   vue/no-unused-vars
  17:28  error  'price' is defined but never used   vue/no-unused-vars
  17:35  error  'title' is defined but never used   vue/no-unused-vars
  17:42  error  'url' is defined but never used     vue/no-unused-vars
  17:57  error  'rating' is defined but never used  vue/no-unused-vars
```
is there any way to properly fix it?

**Reply by zhiyuanzmj** [reply]
You can try upgrading eslint-plugin-vue to v9.20.0.
https://github.com/vuejs/eslint-plugin-vue/releases/tag/v9.20.0

**Reply by AndrewBogdanovTSS** [reply]
I'm on 9.21.0 - didn't help
